src/common/redux/actions/api_key/setKey.ts

```diff
@@ -0,0 +1,28 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { Action } from "readium-desktop/common/models/redux";
+import { IApiKey } from "../../states/api_key";
+
+export const ID = "API_KEY_SET";
+
+export interface Payload extends IApiKey {
+}
+
+export function build(key: string, provider: string, submitted: boolean): Action<typeof ID, Payload> {
```

Comment:
I guess `submitted` is a local component UI state and not a global state needed to be saved to the redux state, What do you think ?